### ISSUE TITLE
Question about the paper: how is Unlearn Success computed exactly?

### ISSUE THREAD

**Issue by molereddy** [zjunlp/KnowUnDo#2]
![image](https://github.com/user-attachments/assets/e5b9a701-9a81-463f-9cb8-71010922a13b)
The above definition is a bit confusing - do you do an exact token-by-token match to see if the generated answer matches the true answer? 
If not, can you elaborate on how the metric is computed?
If yes, won't this be very sensitive to paraphrases?

**Comment by tbozhong** [comment]
Thank you for your attention to our work!

The metric we use, as adopted in previous works such as knowledge unlearning [1] and editing [2], measures performance by comparing the generated answer to the true answer on a token-by-token basis, rather than requiring an exact match. For instance, if the `answer_ids` are `[1, 2, 3, 4]` and the `prediction_ids` are `[2, 2, 3, 4]`, this metric would yield a score of 75%.

We acknowledge that there are limitations in our current metrics for evaluating generation. Therefore, we are exploring alternative metrics, such as evaluations using GPT-4.

[1] [Machine Unlearning of Pre-trained Large Language Models](https://arxiv.org/abs/2402.15159)  
[2] [Editing large language models: Problems, methods, and opportunities](https://arxiv.org/abs/2305.13172)
